### PR TITLE
redis-plus-plus: add version 1.3.7, update hiredis

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **redis-plus-plus/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/redis-plus-plus/all/conanfile.py [hunk 1]
```diff
@@ -67,7 +67,10 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("hiredis/1.0.2")
+        if Version(self.version) < "1.3.0":
+            self.requires("hiredis/1.0.2")
```
Comment (SpaceIm) [inline recipes/redis-plus-plus/all/conanfile.py:71]: hideris 1.1.x is not compatible with redis-plus-plus < 1.3?

Comment (toge) [reply recipes/redis-plus-plus/all/conanfile.py]: @SpaceIm 
Thank you for your comments!
I recheck the compatibility of hiredis, and there is no issue about it.
I try to fix this.

File: recipes/redis-plus-plus/all/conanfile.py [hunk 2]
```diff
@@ -142,6 +142,7 @@ def package_info(self):
             self.cpp_info.components["redis++lib"].requires.append("libuv::libuv")
         if self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.components["redis++lib"].system_libs.append("pthread")
+            self.cpp_info.components["redis++lib"].system_libs.append("m")
```
Comment (SpaceIm) [inline recipes/redis-plus-plus/all/conanfile.py:145]:
```suggestion
            self.cpp_info.components["redis++lib"].system_libs.extend(["m", "pthread"])
```

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**